rules/prefer-t-throws.js

```diff
@@ -0,0 +1,149 @@
+'use strict';
+
+const {visitIf} = require('enhance-visitors');
+const createAvaRule = require('../create-ava-rule');
+const util = require('../util');
+
+// This function checks if there is an AwaitExpression, which is not inside another function.
```

Comment:
Maybe you could find an `AwaitExpression`, then iterate upwards through its parent scopes until you encounter a function, and then check the function.

https://eslint.org/docs/developer-guide/scope-manager-interface

Reply:
What about the `ReturnStatement`? I doubt there's anything simpler than that in the end.

Reply:
Do you have a realistic example of when someone would actually use `return` in a try/catch when testing an error?

Reply:
No, I don't have one. It's pretty hard to find a realistic example but I have a feeling that there might be some code written that way.

Reply:
> I have a feeling that there might be some code written that way.

Rule logic should be based on test fixtures, not feelings. My recommendation to simplify the logic here stands.